jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/core/datagrip/auth/SecretsManagerAuth.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.datagrip.auth
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.intellij.credentialStore.Credentials
+import com.intellij.database.access.DatabaseCredentials
+import com.intellij.database.dataSource.DatabaseAuthProvider
+import com.intellij.database.dataSource.DatabaseAuthProvider.AuthWidget
+import com.intellij.database.dataSource.DatabaseConnectionInterceptor.ProtoConnection
+import com.intellij.database.dataSource.DatabaseCredentialsAuthProvider
+import com.intellij.database.dataSource.LocalDataSource
+import com.intellij.openapi.project.Project
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.future.future
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.credentials.ConnectionSettings
+import software.aws.toolkits.jetbrains.core.datagrip.getAwsConnectionSettings
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.resources.message
+import java.util.concurrent.CompletionStage
+
+data class SecretsManagerConfiguration(
+    val connectionSettings: ConnectionSettings,
+    val secretId: String
```

Comment:
Due to the way the API works both are accepted, I didn't want to scope it down to just arn since typing secret name into the box seems more intuitive to me

Reply:
can we clarify to users that we'll accept either?

Reply:
Yep, changed to "Secret Name/ARN" from "Secret ID"